### PR TITLE
feat(providers): add zulip

### LINKED ISSUE #4380: [NV-2960] 🚀 Feature: Add Zulip Chat Provider
### 🔖 Feature description

Use the [Zulip docs](https://zulip.com/api/send-message) and the [zulip js package](https://github.com/zulip/zulip-js) to create a chat provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

give more providers to the community

### ✌️ How do you aim to achieve this?

community contribution

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2960](https://linear.app/novu/issue/NV-2960/🚀-feature-add-zulip-chat-provider)</sub>

### PR DESCRIPTION
**Pull request by michaldziuba03**
### What change does this PR introduce?

This PR introduces Zulip provider for chat channel.

Docs reference: https://zulip.com/api/incoming-webhooks-overview

### Why was this change needed?

Closes #4380

### Other information (Screenshots)

image

> It works in the same way as Discord chat provider

image

> Logo with square variant visible in workflow editor

image

### CODE REVIEW

File: providers/zulip/src/lib/zulip.provider.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  async sendMessage(data: IChatOptions): Promise<ISendMessageSuccessResponse> {
+    await this.axiosInstance.post(data.webhookUrl, {
+      text: data.content,
+    });
+
+    return {
+      date: new Date().toISOString(),
+    };
+  }
```
Comment (michaldziuba03) [inline providers/zulip/src/lib/zulip.provider.ts:25]: Zulip doesn't return any message id either in response body or in headers.

Response shape from the webhook:

```json
{
	"result": "success",
	"msg": ""
}
```

**Comment by michaldziuba03** [comment]
Docs PR: https://github.com/novuhq/docs/pull/188

**Comment by Cliftonz** [comment]
@michaldziuba03  Can you resolve these conflicts?

**Comment by michaldziuba03** [comment]
@Cliftonz git conflicts resolved

File: providers/zulip/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Novu Zulip Provider
+
+A Zulip chat provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+    FILL IN THE INITIALIZATION USAGE
```
Comment (Cliftonz) [inline providers/zulip/README.md:8]: This needs to be filled in

**Review by Cliftonz — CHANGES_REQUESTED** [review]

**Comment by michaldziuba03** [comment]
@BiswaViraj @Cliftonz

**Review by Cliftonz — APPROVED** [review]

**Pull request merged**